lib/mini_repo/api_router.ex

```diff
@@ -89,6 +213,34 @@ defmodule MiniRepo.APIRouter do
     end
   end
 
+  def get_data_dir(repo) do
+    repo =
+      Application.get_env(:mini_repo, :repositories)
+      |> Enum.filter(fn {key, _} -> key == repo end)
+
+    case repo do
+      [] ->
+        {:error, "Repo not found"}
+
+      [{_key, opts}] ->
+        {_, [root: {_, dir}]} = opts[:store]
+        {:ok, dir}
+
+      _ ->
+        {:error, "Duplicate repos defined"}
+    end
+  end
+
+  defp is_configured_on_demand?() do
+    Application.get_env(:mini_repo, :repositories)
+    |> Enum.filter(fn
+      {:hexpm_mirror, x} -> Keyword.has_key?(x, :on_demand)
```

Comment:
This particular bit assumes too much about names. See the code where it bifurcates into `Mirror` structs [here](https://github.com/wojtekmach/mini_repo/blob/master/lib/mini_repo/application.ex#L38) and I would suggest making that code smarter than making this code smarter, and not hardcoding a configurable name into the code. Moreover, `is_` is usually reserved for guard clauses; the boolean-ness is implied by the `?` at the end.